solana/solana-ibc/programs/solana-ibc/src/tests.rs

```diff
@@ -151,14 +163,85 @@ fn anchor_test_deliver() -> Result<()> {
         ibc::ClientMsg::CreateClient,
         ibc::MsgEnvelope::Client,
     );
+
+    let serialized_msg = message.try_to_vec().unwrap();
+    let serialized_msg_len = serialized_msg.len() as u32;
+
+    println!("\nCreating Account to store message chunks");
+
+    let blockhash = sol_rpc_client.get_latest_blockhash().unwrap();
+    let mut discriminant = vec![0];
+    let data_size_in_bytes = serialized_msg_len.to_le_bytes().to_vec();
+    discriminant.extend(data_size_in_bytes);
+    discriminant.extend(WRITE_ACCOUNT_SEED.to_vec());
```

Comment:
You don’t need to convert everything to Vec.  First of all Vec can be
extended from a slice but more importantly you can use concat:

```suggestion
    let data_size_in_bytes = .to_vec();
    let instruction = [
        &[0][..],
        &serialized_msg_len.to_le_bytes()[..],
        &WRITE_ACCOUNT_SEED[..],
    ].concat();
```